R/checkers.R

```diff
@@ -1,12 +1,22 @@
 #' Check that all dates in dataframe passed in are before a specified date
 #'
+#' @description
+#' This function is specifically meant to ensure that the data in the `df`
+#' specified does nto contain data time-stamped beyond the `max_date`. The
+#' intended use-case for this is to ensure that one doesn't accidentally
+#' pass in data that extends beyond the forecast date, as ideally the user
+#' is providing vintaged "as of" datasets or at the very least is filtering the
+#' data so that they are not including in their inference data that was
+#' made available after the forecast was made.
+#'
+#'
 #' @param df dataframe with `date` column
 #' @param max_date string indicating the maximum date in ISO8601 convention
 #' e.g. YYYY-MM-DD
 #' @param call Calling environment to be passed to the type checker
 #'
 #' @return NULL, invisibly
-check_date <- function(df, max_date, call = rlang::caller_env()) {
+check_date_logic <- function(df, max_date, call = rlang::caller_env()) {
```

Comment:
Should this definitely take in the whole data frame versus take in that columns, i.e.

```suggestion
check_date_logic <- function(date_vector, max_date, call = rlang::caller_env()) {
```

Reply:
I guess since we have the other checks for the date column being present this is fine!